subprojects/gradle/buildchecks/src/main/kotlin/com/avito/android/plugin/build_param_check/BuildParamCheckPlugin.kt

```diff
@@ -70,11 +71,25 @@ open class BuildParamCheckPlugin : Plugin<Project> {
             group = "verification"
             description = "Detects dynamic dependencies"
         }
-        project.tasks.register("checkBuildEnvironment") {
+        val checkBuildEnvironment = project.tasks.register("checkBuildEnvironment") {
             it.group = "verification"
             it.description = "Check typical build problems"
             it.dependsOn(checkAndroidSdk, checkGradleDaemonTask, dynamicDependenciesTask)
         }
+        if (isMac()) {
+            val task = project.tasks.register<OsxLocalhostResolvingTask>("checkOsxLocalhostResolving") {
+                outputs.upToDateWhen { false }
```

Comment:
It is indicator of wrong gradle usage. I think task can be up-to-date when it has inputs and outputs

Reply:
Agree
```
Task ':checkOsxLocalhostResolving' is not up-to-date because:
  Task has not declared any outputs despite executing actions.
```